ApplicationRequirements.txt

```diff
@@ -18,42 +33,107 @@
-		-> Clicking a second time hides the "New Delivery Address" Text Box and the "Confirm New Deliver Address" Button
+	"Deliver To" Text Box
+	"Delivery Address" Text Box
 	"Delivery Instructions" Text Box (Optional: For preferred delivery time and other special instructions)
+	"Pickup From" Text Box
 	"Pickup Address" Text Box
 	"Pickup Instructions" Text Box (Optional: For preferred pickup time and other special instructions)
+	"What is Being Delivered" Text Box
 	"Upload Files" Button (Optional: For uploading receipt/confirmation for online order)
 		-> Opens "File Explorer" to choose file
-	"Submit" Button
-		-> Takes user to "Confirmation" Screen
+	"Checkout" Button
+		-> Takes user to "Checkout" Screen
 	"Cancel" Button
-		-> Takes user to "Home" Screen
+		-> Displays "Confirm Cancel" Pop-up
+		-> Displays "Confirm Cancel" Button
+			-> Takes user to "Home" Screen
 		
-Confirmation Screen
-	Displays "Recipient" Text
+"Checkout" Screen
+	Displays "Deliver To" Text
 	Displays "Delivery Address" Text
 	Displays "Delivery Instructions" Text
+	Displays "Pickup From" Text
 	Displays "Pickup Address" Text
-	Displays "Pickup Instructions" Text
\ No newline at end of file
+	Displays "Pickup Instructions" Text
+	Displays "What is Being Delivered" Text
+	Displays "Uploaded File" Text
+	"Confirm Order" Button
+		-> Takes user to "Confirmation" Screen
+		-> Adds request to the "Request Queue"
+		-> Sets the request status to "Awaiting Driver"
+		-> Adds request to the user's "Request History"
+	"Update Order" Button
+		-> Takes user to "Update Request" Screen
+		
+"Confirmation" Screen
+	Displays "Confirmation" Text
+	"Back to Home" Button
+		-> Takes user back to the "Home" Screen
+
+"Update Request" Screen
+	"Deliver To" Text Box
+		-> Has previous "Deliver To" information in it
+	"Delivery Address" Text Box
+		-> Has previous "Delivery Address" information in it
+	"Delivery Instructions" Text Box (Optional: For preferred delivery time and other special instructions)
+		-> Has previous "Delivery Instructions" information in it
+	"Pickup From" Text Box
+		-> Has previous "Pickup From" information in it
+	"Pickup Address" Text Box
+		-> Has previous "Pickup Address" information in it
+	"Pickup Instructions" Text Box (Optional: For preferred pickup time and other special instructions)
+		-> Has previous "Pickup Instructions" information in it
+	"Upload Files" Button (Optional: For uploading receipt/confirmation for online order)
+		-> Opens "File Explorer" to choose file
+		-> Has previous file in it
+	"Update" Button
+		-> Updates request information
+		-> Takes user to "Confirmation" Screen
+	"Cancel" Button
+		-> Retains request information
+		-> Takes user to "Checkout" Screen if they came from the "Checkout" Screen
+		-> Takes user to "My Requests" Screen if they came from the "My Requests" Screen
+		
+"Requests Available for Pickup" Screen
+	Shows the "Request Queue" of all submitted requests in a certain mile radius
```

Comment:
I agree, but more so as an extra thing that we add in after everything else is done.

Reply:
> we can have sorting options, so drivers can sort the list of orders by time or distance etc

Based on Anas's question for adding delivery time and pickup time, this means it will be added, so we should add it to the requirements doc.